test/subset-band-name/utilities.py

```diff
@@ -0,0 +1,118 @@
+""""
+Common utility functions used by the subset-band-name regression tests.
+"""
+
+import os
+from pyhdf.SD import SD, SDC
+import numpy
+
+from harmony import Client, Request
+from harmony.harmony import ProcessingFailedException
+
+
+def submit_and_download(
+    harmony_client: Client, request: Request, file_indicator: str
+) -> str:
+    """Submit a Harmony request via a `harmony-py` client. Wait for the
+    Harmony job to finish, then download the results to the specified file
+    path.
+
+    """
+    downloaded_filenames = []
+    output_filename = None
+
+    try:
+        job_id = harmony_client.submit(request)
+
+        print(f"Job ID: {job_id}")
+
+        for filename in [
+            file_future.result()
+            for file_future in harmony_client.download_all(job_id, overwrite=True)
+        ]:
+
+            print(f'Downloaded: {filename}')
+            downloaded_filenames.extend([filename])
+
+            # make a copy to /workdir/output (debugging)
+            # output_filename = f'/workdir/output/{filename.split("/")[-1]}'
+            # os.system(f'cp {filename} {output_filename}')
+
+        for filename in downloaded_filenames:
+            if file_indicator in filename:
+                output_filename = filename
+                print(f'Saved output to: {output_filename}')
+            else:
+                print(f'Could not save output to: {filename}')
+
+    except ProcessingFailedException as exception:
+        print_error('Harmony request failed to complete successfully.')
+        raise exception
+
+    return output_filename
+
+
+def get_dim_sizes(file: str) -> list[int]:
+    ydim = None
+    xdim = None
+    file_SD = SD(file, SDC.READ)
+    datasets = file_SD.datasets()
+    if len(datasets) > 0:
+        dataset_names = datasets.keys()
+        for name in dataset_names:
+            dataset = file_SD.select(name)
+            dims = dataset.dimensions(full=1)
+            ydim = list(dims.values())[0][0]
+            xdim = list(dims.values())[1][0]
+
+    file_SD.end()
+    return [ydim, xdim]
+
+
+def get_sds_data(file: str):
+    sds_data = None
+    file_SD = SD(file, SDC.READ)
+    datasets = file_SD.datasets()
+
+    if len(datasets) > 0:
+        dataset_names = datasets.keys()
+        for name in dataset_names:
+            if name == 'Cloud_Mask':
```

Comment:
I think you have a problem here.  

When this function is run against the reference_data, `name == 'Cloud_Mask'` never evaluates to true. For the reference file provided:

If you print datasets on L75 you get 
```
{'Latitude': (('10*nscans:MODIS_SWATH_Type_L1B', 'Max_EV_frames:MODIS_SWATH_Type_L1B'), (2030, 1354), 5, 0),
 'Longitude': (('10*nscans:MODIS_SWATH_Type_L1B', 'Max_EV_frames:MODIS_SWATH_Type_L1B'),  (2030, 1354), 5, 1)}
```
and if you print name on L80 you get 
```
name: Latitude
name: Longitude
```
So you never actually get Data and this just returns `None`, so your comparison is just
```
>>> numpy.array_equal(None, None)
True
```

Reply:
I'm working on this now locally.

Reply:
Got it working. pyhdf did not recognize Band_250M as a dataset. Switched the variable to `EV_250_Aggr500_RefSB` which is an actual dataset.